actix-web-actors/src/ws.rs

```diff
@@ -40,10 +57,13 @@ where
 /// This function returns handshake `HttpResponse`, ready to send to peer.
 /// It does not perform any IO.
 ///
-// /// `protocols` is a sequence of known protocols. On successful handshake,
-// /// the returned response headers contain the first protocol in this list
-// /// which the server also knows.
-pub fn handshake(req: &HttpRequest) -> Result<HttpResponseBuilder, HandshakeError> {
+/// `protocols` is a sequence of known protocols. On successful handshake,
+/// the returned response headers contain the first protocol in this list
+/// which the server also knows.
+pub fn handshake(
+    req: &HttpRequest,
+    protocols: &Vec<&str>,
```

Comment:
this is breaking change, let's use new function

Reply:
also, should this be `Option<&[str]>`?

Reply:
it would make no sense to pass a `Some([])`, so i though to use the empty vec to not use a protocol